semantics/psc-trees.adb

```diff
@@ -80,6 +80,12 @@ package body PSC.Trees is
       null;
    end Set_Nth_Operand;
 
+   function Kind (T : Tree) return Tree_Kind_Enum is
+   begin
```

Comment:
No need for this body if you declare the function abstract.  I.e.:
function Kind(T : Tree) return Tree_Kind_Enum is abstract;